src/components/Select/Select/Select.tsx

```diff
@@ -56,33 +54,47 @@ export const Select = (props: TSingleSelectPropTypes): JSX.Element | null => {
     labelAddons,
     tooltipAddons
   } = props
+
   const scrollRef = useRef<HTMLDivElement>(null)
   const inputRef = useRef<HTMLInputElement | null>(null)
-  const { scrollHeight } = useGetElemSizes(scrollRef.current)
-  const [isOpen, setIsOpen] = useState<boolean>(false)
   const containerRef = useRef<HTMLDivElement>(null)
+  const [isOpen, setIsOpen] = useState<boolean>(false)
   const [searchValue, setSearchValue] = useState<string>('')
   const [dropdownRef, setDropdownRef] = useState<HTMLDivElement | null>(null)
+  const [isDynamicSearchEnabled, setIsDynamicSearchEnabled] = useState<boolean>(false)
+
   const currentSelection = (value as TItemValue) || selectedItem
   const [selectedOption, setSelectedOption] = useState<TSelectOption | null>(null)
 
-  const isWithSearch = withSearch && options.length > SELECTED_VISIBLE_MIN_COUNT
+  const isWithSearch = withSearch && isDynamicSearchEnabled
 
-  const setCurrentSelectedLabel = useCallback(() => {
+  const setCurrentSelectedLabel = () => {
     const selectedItem = options.find((item) => item.value === currentSelection) as TSelectOption
     setSelectedOption(selectedItem)
-  }, [currentSelection, options])
+  }
 
   const leftIconProps = selectedOption?.optionLeftIcon?.Component
     ? {
         Component: selectedOption?.optionLeftIcon?.Component
       }
     : null
 
+  const handleScrollHeightCheck = useCallback(() => {
```

Comment:
@KaroGhulyan jan remove 
```
  useEffect(() => {
    if (isOpen) {
      handleScrollHeightCheck()
    }
  }, [isOpen, handleScrollHeightCheck])
  ``` 
  instead of `handleScrollHeightCheck` and `isDynamicSearchEnabled, setIsDynamicSearchEnabled` use useMemo like this 
  ```
  const isDynamicSearchEnabled = useMemo(() => {
     if (isOpen) {
        ...
     };
     
     return false
  }, [isOpen, ]dropdownRef)
  ```

Reply:
@rispiryan jan I didn`t understand,  It will work without useCallback and be more optimal.

Reply:
@valodyapapikyan jan i think you answered under wrong comment. am i right?